docs/python/generators/doxygen.py

```
@@ -119,21 +182,23 @@ def embed_doc(nested, fos, names_to_search_examples=[], printed_example_pairs=se
                     print(file=fos)
-                    print(f"    [{ex['title']}]({ex['id']}.md) - {ex.get('description','')}", file=fos)
+                    print(f"    {_example_title(ex)}", file=fos)
                     print(file=fos)
-                    print(f"    ```{extension}{hl_attr}", file=fos)
-                    for line in ex['snippet'].splitlines():
-                        print(f"    {line}", file=fos)
-                    print(f"    ```", file=fos)
+                    if snippet:
+                        render_lines = snippet.splitlines()
+                        first_nonblank = 0
+                        for l in render_lines:
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)